clab/authz_keys.go

```diff
@@ -18,24 +19,35 @@ import (
 )
 
 const (
-	pubKeysGlob = "~/.ssh/*.pub"
+	pubKeysGlobPart = ".ssh/*.pub"
 	// authorized keys file path on a clab host that is used to create the clabAuthzKeys file.
-	authzKeysFPath = "~/.ssh/authorized_keys"
+	authzKeysFPathPart = ".ssh/authorized_keys"
 )
 
 // CreateAuthzKeysFile creats the authorized_keys file in the lab directory
 // if any files ~/.ssh/*.pub found.
 func (c *CLab) CreateAuthzKeysFile() error {
 	b := new(bytes.Buffer)
 
-	p := utils.ResolvePath(pubKeysGlob, "")
+	userId, isSet := os.LookupEnv("SUDO_UID")
+	if !isSet {
+		userId = "0" // default to root
+	}
+
+	// lookup user to figure out Home Directory
+	u, err := user.LookupId(userId)
+	if err != nil {
+		return err
+	}
+
+	p := utils.ResolvePath(filepath.Join(u.HomeDir, pubKeysGlobPart), "")
```

Comment:
Hi @steiler 
this is a good idea, but I'd rather do this centrally in `utils.ResolvePath`
There you can use the code branch that resolves the `~` char to the user's home directory. You can just use the code block that looks up the SUDO_UID there to identify which user's home dir should it be resolved to

Reply:
Alright, yes, that definetely makes sense. Thanks!